src/cookies.rs

```diff
@@ -47,18 +48,18 @@ impl<'a, 'b, 'k, D> Plugin<Response<'a, D>> for CookiePlugin
 where D: KeyProvider {
     type Error = ();
 
-    fn eval(res: &mut Response<'a, D>) -> Result<CookieJar<'static>, ()> {
+    fn eval(res: &mut Response<'a, D>) -> Result<CookieJar, ()> {
         // Schedule the cookie to be written when headers are being sent
         res.on_send(|res| {
             let header = {
                 let jar = res.get_ref::<CookiePlugin>().unwrap();
-                header::SetCookie::from_cookie_jar(jar)
+                // header::SetCookie::from_cookie_jar(jar)
+                header::SetCookie(jar.delta().map(|c| c.to_string()).collect())
             };
             res.set(header);
         });
 
-        let key = res.server_data().key();
-        Ok(CookieJar::new(&key.0))
+        Ok(CookieJar::new())
```

Comment:
Does the new CookieJar no longer support encryption? I feel like if we're to drop that as a feature then we should at least change some of the documentation around this library or implement those features using another library, perhaps behind a feature?

Reply:
I don't fully recall now why I had to remove this. Looking at the docs for the cookie crate, it looks like signed cookies are supported, but the api has changed. When I made this patch, I was trying to get our project working with nickel-0.10. We aren't using encrypted cookies, so I punted when it turned complicated.

It may be best to regard this pull request as more documentation of some of the issues I discovered.

Reply:
Done in latest version of PR

Reply:
I've had more time to look into this. The API changed from 0.2 to 0.8 and on. Now CookieJar is not encrypted or signed, but cookies in the jar can be by the private and signed methods of CookieJar. Basically you would use cookies_mut to get a mutable jar, and use those methods for secure access. I believe we don't need to do anything more to support secure cookies.

Reply:
My previous comment only applies to 0.10. 0.8 doesn't have this feature, and signing/encryption is on a per cookie basis.

I am working on updating the example. At the same time I think I should update to cookie-0.10. Your thoughts?

Reply:
I was wrong, 0.8 supports this feature. I have updated the example to use the cookie crate's new approach to secure access. In the process I removed the AllowMutCookies trait, since the secured access require mutable access to the Request cookies.

There might be some room for a cleaner API, but I think that ought to be part of a different PR.